src/Aspire.Hosting/Extensions/ContainerResourceBuilderExtensions.cs

```diff
@@ -65,6 +66,24 @@ public static IResourceBuilder<ContainerResource> AddContainer(this IDistributed
         return builder.WithAnnotation(annotation);
     }
 
+    /// <summary>
+    /// Adds a named volume mount to the container resource where the name is generated from
+    /// the <see cref="IHostEnvironment.ApplicationName"/> and <see cref="Resource.Name"/>.
+    /// </summary>
+    /// <typeparam name="T">The resource type.</typeparam>
+    /// <param name="builder">The resource builder.</param>
+    /// <param name="target">The target path in the container.</param>
+    /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<T> WithPersistentVolume<T>(
+        this IResourceBuilder<T> builder,
+        string target)
+        where T : ContainerResource
+    {
+        var source = $"{builder.ApplicationBuilder.Environment.ApplicationName}.{builder.Resource.Name}.data";
```

Comment:
What error do you get when this is called multiple times?

Reply:
I can check real quick, but I'd assume it would be Docker specific.

How does Aspire think about this problem?  Does it try and prevent a user from misusing an underlying API?  Or does it bubble everything up?

Reply:
>What error do you get when this is called multiple times?

**same `target`**
Using the same `target` mount point from the same `ContainerResource` will cause the docker container to fail to start.  

This should be the docker equivalent:
```
>docker run -d \
   --name example-multiple-mounts \
    -e PGDATA=/var/lib/postgresql/data/pgdata 
    --mount 'type=volume,src=test-multiple,dst=/var/lib/postgresql,volume-driver=local' \
    --mount 'type=volume,src=test-multiple,dst=/var/lib/postgresql,volume-driver=local' \
    postgres
docker: Error response from daemon: Duplicate mount point: /var/lib/postgresql.
See 'docker run --help'. 
```
**different `target`**
This works fine - as far as I can tell, the same volume just has multiple mount points inside the container.  Whether or not it's useful is another story.

```csharp
var catalogDb =
    builder
        .AddPostgresContainer("postgres")
        .WithEnvironment("PGDATA", "/var/lib/postgresql/data/pgdata")
        .WithPersistentVolume("/var/lib/postgresql/data1")
        .WithPersistentVolume("/var/lib/postgresql/data2")
        .WithPersistentVolume("/var/lib/postgresql/data3")  // <------ this runs fine
        .AddDatabase("catalogdb");
```

Reply:
I think that's for us to decide. If this is a singleton instance then we should make it noop if you call it more than once. Otherwise, we should force you to provide a name and we can setup a prefix.